File: backend/models/image.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

type Image struct {
	ID        primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	UserID    primitive.ObjectID `bson:"userID" json:"userID"`
	Path      string             `bson:"path" json:"path"`
	Imagename string             `bson:"imagename" json:"imagename"`
	CreatedAt primitive.DateTime `json:"createdAt"`
}

File: frontend/lib/models/image.dart
class Image {
  final String? id;
  final String? path;
  final String? imagename;

  Image({
    this.id,
    this.path,
    this.imagename,
  });

  factory Image.fromJson(Map<String, dynamic> json) {
    return Image(
      id: json['id'],
      path: json['path'],
      imagename: json['imagename'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'path': path,
      'imagename': imagename,
    };
  }
}
